src/util.js

```diff
@@ -276,6 +276,45 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+var _hookIdCounter = 0;
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
```

Comment:
I am not sure if reflecting property keys and looping through them for every method call is acceptable even for debugging mode. I need to think of a better way to do this.